ui/tests/unit/routes/vault/cluster/oidc-callback-test.js

```diff
@@ -207,4 +208,121 @@ module('Unit | Route | vault/cluster/oidc-callback', function (hooks) {
       'namespace is parsed correctly'
     );
   });
+
+  module('getParamsForCallback helper fn', function () {
+    test('it parses params correctly with regular inputs and no namespace', function (assert) {
+      const qp = {
+        state: 'my-state',
+        code: 'my-code',
+        path: 'oidc-path',
+      };
+      const searchString = `?code=my-code&state=my-state`;
+      const results = getParamsForCallback(qp, searchString);
+      assert.deepEqual(results, { source: 'oidc-callback', ...qp });
+    });
+
+    test('it parses params correctly regular inputs and namespace param', function (assert) {
+      const params = {
+        state: 'my-state',
+        code: 'my-code',
+        path: 'oidc-path',
+        namespace: 'my-namespace',
+      };
+      const results = getParamsForCallback(params, '?code=my-code&state=my-state&namespace=my-namespace');
```

Comment:
The string is basically the expected URL, and the namespace in the params is what the Ember route parses from the URL. So we're not testing differences here, just trying to match the real world scenario :)

Reply:
Ah - so many params! I got mixed up because I thought the queryString passed to `getParamsForCallback` was always whatever was from `window.location.search` and I never saw `namespace=` as part of that search string